near-sdk/src/types/event.rs

```diff
@@ -0,0 +1,95 @@
+pub trait StandardEvent {
+    fn format(&self) -> String;
+    fn emit(&self);
```

Comment:
This could have a default implementation to prevent devs from having to implement it manually if they choose.
```suggestion
    fn emit(&self) {
        crate::env::log_str(&format!("EVENT_JSON:{}", &standard_event.format()));
    }
```